<launch>
    <param name="img_proc_bgr_topic" type="string" value="camera/image_raw/image_rect_color" />
    <param name="img_proc_depth_topic" type="string" value="camera/aligned_depth_to_color/image_raw/image_rect_color" />
    <param name="processed_image_topic" type="string" value="barbot/processed" />
    <param name="cup_detection_topic" type="string" value="barbot/cup_detections" />
    <param name="target_position_topic" type="string" value="barbot/target_position" />
    <param name="target_gripper_state_topic" type="string" value="barbot/target_gripper_state" />
    <param name="current_pose_topic" type="string" value="barbot/current_pose" />

    <!-- Camera -->
    <!-- <include file="$(find realsense2_camera)/launch/rs_camera.launch"/> -->
    <node name ="detector_node" pkg="bar_bot" type="detector_node.py" output="screen" required="true"/>

    <!-- Hebi interaction/kinematics -->

    <node name="hebiros_node" pkg="hebiros" type="hebiros_node" respawn="false" output="screen">
      <remap from="/hebiros/all/command/joint_state" to="/command_joint_state"/>
    </node>
    <node name="copy_topic" pkg="topic_tools" type="relay"
      args="/hebiros/all/feedback/joint_state /feedback_joint_state"/>
    <node name="copy_topic2" pkg="topic_tools" type="relay"
      args="/hebiros/all/command/joint_state /command_joint_state"/>

    <node name="mobility" pkg="bar_bot" type="mobility_server" output="screen"/>

    <!-- Commander -->
    <node name="commander" pkg="bar_bot" type="commander_node" output="screen" />
    <node name="rqt" pkg="rqt_gui" type="rqt_gui"/>

    <!-- Generic ROS/Visualization stuff -->
    <!-- <param name="robot_description" textfile="$(find bar_bot)/urdf/basic_arm.urdf"/> -->
    <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find bar_bot)/rviz/viewrobot.rviz"/> -->
    <!-- <node name="rsp" pkg="robot_state_publisher" type="robot_state_publisher"> -->
    <!--     <param name="ignore_timestamp" value="true"/> -->
    <!-- </node> -->

    <!-- <node name="jsp" pkg="joint_state_publisher" type="joint_state_publisher"> -->
    <!--   <param name="use_gui" value="true"/> -->
    <!-- </node> -->

    <!-- Arduino Serial -->
    <!-- <node name="arduino_serial" pkg="rosserial_python" type="serial_node.py" args="_port:=/dev/ttyACM0" /> -->
</launch>
